Contributing.md

```diff
@@ -14,40 +14,73 @@ Conduct](CODE_OF_CONDUCT.md).
 Rustfmt requires a nightly compiler. Replace all invocations of `cargo` by
 `cargo +nightly` if necessary.
 
-At all times the environment variables `CFG_RELEASE_CHANNEL` and `CFG_RELEASE` must
-be set to either `nightly` or `beta`. Both should be set to the same value. The
-project supports [cargo-make](https://github.com/sagiegurari/cargo-make) which
-automatically sets these variables for you:
+At all times the environment variables `CFG_RELEASE_CHANNEL` and `CFG_RELEASE` must be set. `CFG_RELEASE_CHANNEL` must be set to either `nightly` or `beta`, and `CFG_RELEASE` should be set to your corresponding Rust version. The
+project supports and recommends [cargo-make][cargo-make] when working on rustfmt, because `cargo-make` automatically sets these variables for you
 
+The available tasks are listed in `Makefile.toml`.
+
+You can alternatively use `cargo` directly, but you'll have to set the `CFG_RELEASE` and `CFG_RELEASE_CHANNEL` environment variables and also provide the corresponding features.
+
+For example:
+```sh
+export CFG_RELEASE=1.45.0-nightly
+export CFG_RELEASE=nightly
 ```
-cargo +nightly make test
+(Windows users can use `set` to specify the environment variable values)
+
+## Building
+To build with `cargo make`:
+
+```sh
+cargo make build
 ```
 
-The available tasks are listed in `Makefile.toml`.
+Or alternatively with `cargo` directly:
+```sh
+cargo build --all-features
```

Comment:
Maybe worth mentioning the `export` commands before this line given the next line sets the env variables too

Reply:
Hmmm. Obviously these aren't mutually exclusive and a rustfmt contributor/developer that has elected to eschew the usage of `cargo make` could set the environment variables, but also override on certain invocations.

A very real use case would be:
```sh
CFG_RELEASE_CHANNEL=beta cargo test --all-features
```

If the either/or theme isn't already clear, then I'm not convinced swapping the order impacts the clarity one way or the other. Maybe some additional text above on the vars and inline on the samples

Reply:
Good point, this makes sense. On the other hand for a new contributor it may not be immediately obvious why the second line should be used instead of the first. Maybe the “or” comment could be extended to mention the use case you brought up, of testing a particular env?